esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+See also [if-source](#if-source).
+
+### store-value
+```xml
+    <store-value
+        name="String value"
+        optional="Boolean value"
+        select="XPath String value"
+        trace="String value"
+        xpath_name="XPath String value"
+    />
+```
+
+Add a name-value pair to the *store* section of the script context. The name must be valid as an XML attribute. The value must be encoded for storage as an attribute value.
+
+| Property | Count | Description |
+| - | - | - |
+| @name | 0..1 | String label identifying a value. |
+| @optional | 0..1 | Boolean flag indicating whether script syntax errors are fatal (*false*) or merely generate warnings (*true*). Defaults to *false*. |
+| @select | 1..1 | XPath evaluating to an option value. |
+| @trace | 0..1 | Label used in trace log output messages. If omitted or empty, the element name is used. |
+| @xpath_name | 0..1 | Xpath evaluating to a value identifier. |
+
+- While `@name` and `@xpath_name` are both individually optional, at least one must be given. If both are given, `@xpath_name` is used.
+
+### target
+```xml
+    <target
+        optional="Boolean value"
+        required="Boolean value"
+        trace="String value"
+        xpath="XPath node expression"
+    />
+```
+
+Temporarily change the output document's effective root node to an existing output document node. Relative paths specified in child operations are relative to the indicated node. The effective root resets to its previous value upon completion of child processing.
+
+| Property | Count | Description |
+| - | - | - |
+| @optional | 0..1 | Boolean flag indicating whether script syntax errors are fatal (*false*) or merely generate warnings (*true*). Defaults to *false*. |
+| @required | 0..1 | Boolean flag indicating whether a failure to uniquely identify an existing output node terminates processing (*true*) or skips processing child operations (*false*). |
+| @trace | 0..1 | Label used in trace log output messages. If omitted or empty, the element name is used. |
+| @xpath | 1..1 | XPath expression evaluating to an existing output document node. |
+
+See also [ensure-target](#ensure-target) and [if-target](#if-target).
+
+### variable
+```xml
+    <variable name="String value"
+              select="XPath expression"
+    />
+```
+
+Define an immutable variable for use by the script. It is in scope for any sibling or descendent elements.
+
+| Property | Count | Description |
+| - | - | - |
+| @name | 1..1 | Required name of the variable. |
+| @select | 0..1 | Evaluation of this XPath expression gives the default value of the variable. |
+
+> Note: The `variable` operation cannot be a child of `choose`. The script will parse but could throw an error during runtime. If the variable is referenced inside an XPath the error **Could not evaluate XPATH '*...offending xpath..*'** is thrown. A future update will handle this error condition more robustly.
```

Comment:
I understand the sentiment behind this comment. Maybe add an explicit note with `choose` indicating that all child elements but `when` and `otherwise` are ignored during load - parsing succeeds and runtime outcome will not be as expected.

Reply:
fixed